contracts/cw20-streams/src/contract.rs

```diff
@@ -0,0 +1,575 @@
+        let start_time = env.block.time.plus_seconds(100).seconds();
+        let end_time = env.block.time.plus_seconds(300).seconds();
+
+        let msg = ExecuteMsg::Receive(Cw20ReceiveMsg {
+            sender,
+            amount,
+            msg: to_binary(&ReceiveMsg::CreateStream {
+                recipient,
+                start_time,
+                end_time,
+            })
+            .unwrap(),
+        });
+        execute(deps.as_mut(), env.clone(), info, msg).unwrap();
+
+        let msg = QueryMsg::GetStream { id: 1 };
+        let res = query(deps.as_ref(), mock_env(), msg).unwrap();
+        let stream: Stream = from_binary(&res).unwrap();
+
+        assert_eq!(
+            stream,
+            Stream {
+                owner: Addr::unchecked("Alice"),
+                recipient: Addr::unchecked("Bob"),
+                amount,
+                claimed_amount: Uint128::new(0),
+                start_time,
+                rate_per_second: Uint128::new(1),
+                end_time
+            }
+        );
+
+        // Stream has not started
+        let mut info = mock_info(MOCK_CONTRACT_ADDR, &[]);
+        info.sender = Addr::unchecked("Bob");
+        let msg = ExecuteMsg::Withdraw { id: 1 };
+        let err = execute(deps.as_mut(), env, info, msg).unwrap_err();
+        assert_eq!(err, ContractError::NoFundsToClaim {});
+
+        // Stream has started so tokens have vested
+        let msg = ExecuteMsg::Withdraw { id: 1 };
+        let mut info = mock_info(MOCK_CONTRACT_ADDR, &[]);
+        let mut env = mock_env();
+        info.sender = Addr::unchecked("Bob");
+        env.block.time = env.block.time.plus_seconds(150);
+        let res = execute(deps.as_mut(), env, info, msg).unwrap();
+        let msg = res.messages[0].clone().msg;
+
+        assert_eq!(
+            msg,
+            CosmosMsg::Wasm(WasmMsg::Execute {
+                contract_addr: String::from(MOCK_CONTRACT_ADDR),
+                msg: to_binary(&Cw20ExecuteMsg::Transfer {
+                    recipient: String::from("Bob"),
+                    amount: Uint128::new(50)
+                })
+                .unwrap(),
+                funds: vec![]
+            })
+        );
+
+        let msg = QueryMsg::GetStream { id: 1 };
+        let res = query(deps.as_ref(), mock_env(), msg).unwrap();
+        let stream: Stream = from_binary(&res).unwrap();
+        assert_eq!(
+            stream,
+            Stream {
+                owner: Addr::unchecked("Alice"),
+                recipient: Addr::unchecked("Bob"),
+                amount,
+                claimed_amount: Uint128::new(50),
+                start_time,
+                rate_per_second: Uint128::new(1),
+                end_time
+            }
+        );
+
+        // Stream has ended so claim remaining tokens
+
+        let mut env = mock_env();
+        env.block.time = env.block.time.plus_seconds(500);
+        let mut info = mock_info(MOCK_CONTRACT_ADDR, &[]);
```

Comment:
I wouldn't use this as the owner. Set an arbitrary owner address... like: `let owner = Addr::unchecked("owner");`